src/io/flutter/editor/ActiveEditorsOutlineService.java

```diff
@@ -186,8 +189,61 @@ public FlutterOutline get(String path) {
    * Gets the most up-to-date {@link FlutterOutline} for the {@param file}.
    */
   @Nullable
-  public FlutterOutline get(VirtualFile file) {
-    return get(file.getCanonicalPath());
+  public FlutterOutline get(@NotNull VirtualFile file) {
+    return pathToOutline.get(file.getCanonicalPath());
+  }
+
+  /**
+   * Gets the up-to-date {@link FlutterOutline} for the {@param file}.
+   *
+   * <p>
+   * If the cached outline is not updated yet, will request an updated outline from the analysis service.
+   */
+  public FlutterOutline request(@NotNull VirtualFile file) {
+    if (isUpToDate(file)) {
+      return get(file);
+    }
+    final String path = file.getCanonicalPath();
+    // If we haven't gotten any data, set up a temporary listener for this file, and wait for a response.
+    // We block to wait with a Lock.
+    final Lock lock = new ReentrantLock();
```

Comment:
We need to avoid blocking the UI thread in IntelliJ. How do we ensure this API isn't called in a blocking way on the UI thread?
An alternate design would just be to assume that an updated outline will arrive at some point if the current outline is out of date at which point the correct code flow can trigger based on the existing event driven API.

Reply:
I have finally found a way of requesting a new line markers pass:

```java
DaemonCodeAnalyzer.getInstance(project).restart();
```

The IntelliJ Dart plugin does this in cases where it receives new data from the dart analyzer already.  I've updated this PR with that data.